tests/pauli/test_conversion.py

```diff
@@ -209,6 +209,19 @@ def test_wire_error(self, pauli):
         ):
             qml.pauli_decompose(hamiltonian, pauli=pauli, wire_order=wire_order)
 
+    @pytest.mark.jax
+    def test_jit(self):
+        """Test that pauli_decompose can be used inside a jit context"""
+        import jax
+
+        @jax.jit
+        def f(m):
+            coeffs, unitaries = qml.pauli_decompose(m, check_hermitian=False).terms()
```

Comment:
What if the user did not include `check_hermitian=False`? Should we add the `is_abstract` check alongside `if check_hermitian` in `pauli_decompose`?

Reply:
`check_hermitian=False` is definitely needed in this version. Let me try adding `is_abstract` along the hermiticity check as well

Reply:
very good suggestion, I was actually running into the same abstract jax array difficulties with the hermiticity check in catalyst. This is definitely the place to fix that as well :100:

Reply:
Now `check_hermitian=False` wouldn't be needed here I think?

Reply:
It is still needed, as this input matrix is purposefully not hermitian (so turning it off would real-lize the coefficients and would be giving the wrong answers). I added another test for hermitian matrices, where `check_hermitian` kwarg is omitted